### PR TITLE
[macos] prepare repository for ARM64 arch

### PR DESCRIPTION
**Pull request by Alexey-Ayupov**
# Description
Prepare repository for upcoming ARM64 arch

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/cocoapods.sh [hunk 1]
```diff
@@ -1,10 +1,18 @@
 #!/bin/bash -e -o pipefail
 
+source ~/utils/utils.sh
```
Comment (mikhailkoliada) [inline images/macos/provision/core/cocoapods.sh:3]: probably we do not need to change this file too

File: images/macos/provision/configuration/shell-change.sh [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+source ~/utils/utils.sh
+arch=$(get_arch)
+
 echo "Changing shell to bash"
 sudo chsh -s /bin/bash $USERNAME
-sudo chsh -s /bin/bash root
\ No newline at end of file
+sudo chsh -s /bin/bash root
+
+# Check MacOS architecture and add HOMEBREW PATH to bashrc
```
[the remark this answers is not in the document]
Comment (Alexey-Ayupov) [reply images/macos/provision/configuration/shell-change.sh]: done

File: images/macos/provision/core/powershell.sh [hunk 3]
```diff
@@ -19,19 +21,36 @@ for module in ${psModules[@]}; do
...
+        else
+            pwsh -command "& {Install-Module $module -Force -Scope AllUsers}"
+        fi
     else
         for version in ${moduleVersions[@]}; do
-            echo " - $version"
-            pwsh -command "& {Install-Module $module -RequiredVersion $version -Force -Scope AllUsers}"
+            # Check MacOS architecture and sudo on Arm64
+            if [[ $arch == "arm64" ]]; then
+                echo " - $version"
```
[the remark this answers is not in the document]
Comment (Alexey-Ayupov) [reply images/macos/provision/core/powershell.sh]: done

File: images/macos/provision/core/dotnet.sh [hunk 4]
```diff
@@ -37,14 +39,25 @@ for DOTNET_VERSION in "${DOTNET_VERSIONS[@]}"; do
...
-    ./dotnet-install.sh --version $ARGS -NoPath
+    ./dotnet-install.sh --version $ARGS -NoPath --arch $arch
 done
 
 rm ./dotnet-install.sh
 
-# dotnet installer doesn't create symlink to executable in /user/local/bin
-# Moreover at that moment /user/local/bin doesn't exist (though already added to $PATH)
-ln -s ~/.dotnet/dotnet /usr/local/bin/dotnet
+echo "creating software link"
```
[the remark this answers is not in the document]
Comment (Alexey-Ayupov) [reply images/macos/provision/core/dotnet.sh]: done

File: images/macos/provision/core/powershell.sh [hunk 5]
```diff
@@ -19,19 +21,36 @@ for module in ${psModules[@]}; do
...
+            else
+                echo " - $version"
+                pwsh -command "& {Install-Module $module -RequiredVersion $version -Force -Scope AllUsers}"
+            fi
         done
     fi
 done
 
+# Cleanup PowerShell permissions (If left it blocks ruby.sh from running correctly since it attmpts to create ~/.local/share/)
+if [[ $arch == "arm64" ]]; then
```
[the remark this answers is not in the document]
Comment (Alexey-Ayupov) [reply images/macos/provision/core/powershell.sh]: done

File: images/macos/provision/core/powershell.sh [hunk 6]
```diff
@@ -19,19 +21,36 @@ for module in ${psModules[@]}; do
     echo "Installing $module module"
     moduleVersions="$(get_toolset_value ".powershellModules[] | select(.name==\"$module\") | .versions[]?")"
     if [[ -z "$moduleVersions" ]];then
-        pwsh -command "& {Install-Module $module -Force -Scope AllUsers}"
+        # Check MacOS architecture and sudo on Arm64
+        if [[ $arch == "arm64" ]]; then
```
[the remark this answers is not in the document]
Comment (tuxity) [reply images/macos/provision/core/powershell.sh:25]: I also needed to add sudo here

File: images/macos/provision/core/rosetta.sh [hunk 7]
```diff
@@ -0,0 +1,15 @@
+  #!/bin/bash -e -o pipefail
```
Comment (mayeut) [inline images/macos/provision/core/rosetta.sh:1]:
```suggestion
#!/bin/bash -e -o pipefail
```

File: images/macos/provision/core/powershell.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (Alexey-Ayupov) [reply images/macos/provision/core/powershell.sh:25]: approach with "sudo" was tested, "sudo" was added

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 8]
```diff
@@ -40,7 +40,7 @@ $languageAndRuntime.AddNodes($(Get-GccVersions))
 $languageAndRuntime.AddNodes($(Get-FortranVersions))
 $languageAndRuntime.AddToolVersion("Julia", $(Get-JuliaVersion))
 $languageAndRuntime.AddToolVersion("Kotlin", $(Get-KotlinVersion))
-if (-not $os.IsVentura) {
+if (-not $os.IsVentura -and -not $os.IsVenturaArm64) {
```
Comment (mikhailkoliada) [inline images/macos/software-report/SoftwareReport.Generator.ps1:43]: @Alexey-Ayupov it should be `-or`?

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 9]
```diff
@@ -241,6 +258,7 @@ $unitTestFramework.AddToolVersion("NUnit", $(Get-NUnitVersion))
 }
 
 # Xcode section
+write-host "Xcode section"
```
Comment (mikhailkoliada) [inline images/macos/software-report/SoftwareReport.Generator.ps1:261]: do we need this write-host?

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 10]
```diff
@@ -256,16 +274,17 @@ $installedSdks.AddTable($(Build-XcodeSDKTable $xcodeInfo))
 
 $installedSimulators = $xcode.AddHeader("Installed Simulators")
 $installedSimulators.AddTable($(Build-XcodeSimulatorsTable $xcodeInfo))
-
+write-host "Xcode section end"
```
Comment (mikhailkoliada) [inline images/macos/software-report/SoftwareReport.Generator.ps1:277]: and this too

File: images/macos/provision/core/dotnet.sh [hunk 11]
```diff
@@ -46,6 +48,8 @@ rm ./dotnet-install.sh
 # Moreover at that moment /user/local/bin doesn't exist (though already added to $PATH)
 ln -s ~/.dotnet/dotnet /usr/local/bin/dotnet
 
+echo "Validate installation"
```
Comment (mikhailkoliada) [inline images/macos/provision/core/dotnet.sh:51]: drop the line

File: images/macos/provision/core/ruby.sh [hunk 12]
```diff
@@ -1,45 +1,56 @@
 #!/bin/bash -e -o pipefail
 source ~/utils/utils.sh
 
+arch=$(get_arch)
+echo $arch
```
Comment (mikhailkoliada) [inline images/macos/provision/core/ruby.sh:5]: drop this line

File: images/macos/provision/core/rubygem.sh [hunk 13]
```diff
@@ -1,18 +1,27 @@
...
 
+arch=$(get_arch)
+
 echo Updating RubyGems...
 gem update --system
 
 gemsToInstall=$(get_toolset_value '.ruby.rubygems | .[]')
 if [ -n "$gemsToInstall" ]; then
     for gem in $gemsToInstall; do
+        if [[ $arch == "arm64" ]]; then
```
Comment (mikhailkoliada) [inline images/macos/provision/core/rubygem.sh:12]: lets drop nomad-cli and leave this file untouched at all.

File: images/macos/provision/core/rubygem.sh [hunk 14]
```diff
@@ -1,18 +1,27 @@
...
+            fi
+        fi
         echo "Installing gem $gem"
         gem install $gem
     done
 fi
 
 # Temporary uninstall public_suffix 5.0 gem as Cocoapods is not compatible with it yet https://github.com/actions/runner-images/issues/6149
-gem uninstall public_suffix -v 5.0.0
+gem uninstall public_suffix -v 5.0.1
```
Comment (mikhailkoliada) [inline images/macos/provision/core/rubygem.sh:25]: also probably we may drop this line

File: images/macos/provision/core/xcode.ps1 [hunk 15]
```diff
@@ -14,8 +14,13 @@ if ([string]::IsNullOrEmpty($env:XCODE_INSTALL_USER) -or [string]::IsNullOrEmpty
 # https://github.com/fastlane/fastlane/pull/18116
 $env:SPACESHIP_SKIP_2FA_UPGRADE = 1
 
-$os = Get-OSVersion
-[Array]$xcodeVersions = Get-ToolsetValue "xcode.versions"
+$ARCH = arch
+if ($ARCH -ne "arm64"){
+    $ARCH = "x64"
+}
+write-host $ARCH
```
Comment (mikhailkoliada) [inline images/macos/provision/core/xcode.ps1:21]: drop this line

File: images/macos/tests/Browsers.Tests.ps1 [hunk 16]
```diff
@@ -1,49 +1,52 @@
-Describe "Chrome" {
+Import-Module "$PSScriptRoot/../helpers/Common.Helpers.psm1"
+$os = Get-OSVersion
+
+Describe "Chrome" -Skip:($os.IsVenturaArm64) {
```
Comment (mikhailkoliada) [inline images/macos/tests/Browsers.Tests.ps1:4]: lets drop skips from `It` calls and only leave to the `Describe` blocks

File: images/macos/tests/System.Tests.ps1 [hunk 17]
```diff
@@ -39,13 +39,13 @@ Describe "Audio device" -Skip:($os.IsVentura) {
...
 
-Describe "Screen Resolution" {
+Describe "Screen Resolution" -Skip:($os.IsVenturaArm64) {
     It "Screen Resolution" {
         system_profiler SPDisplaysDataType | Select-String "Resolution" | Should -Match "1176 x 885|1920 x 1080"
     }
 }
 
-Describe "Open windows" {
+Describe "Open windows" -Skip:($os.IsVenturaArm64) {
```
Comment (mikhailkoliada) [inline images/macos/tests/System.Tests.ps1:48]: skip this test for intel too?

File: images/macos/tests/System.Tests.ps1 [hunk 18]
```diff
@@ -39,13 +39,13 @@ Describe "Audio device" -Skip:($os.IsVentura) {
     }
 }
 
-Describe "Screen Resolution" {
+Describe "Screen Resolution" -Skip:($os.IsVenturaArm64) {
```
Comment (mikhailkoliada) [inline images/macos/tests/System.Tests.ps1:42]: skip this test for intel too?

File: images/macos/tests/System.Tests.ps1 (hunk 17 above)
Comment (mikhailkoliada) [reply images/macos/tests/System.Tests.ps1:48]: @Alexey-Ayupov lets introduce `IsVeertu` check to skip Anka-specific conditions

Comment (Alexey-Ayupov) [reply images/macos/tests/System.Tests.ps1:48]: Done

File: images/macos/tests/System.Tests.ps1 (hunk 18 above)
Comment (Alexey-Ayupov) [reply images/macos/tests/System.Tests.ps1:42]: done

File: images/macos/tests/Browsers.Tests.ps1 (hunk 16 above)
Comment (Alexey-Ayupov) [reply images/macos/tests/Browsers.Tests.ps1:4]: Done

File: images/macos/software-report/SoftwareReport.Generator.ps1 (hunk 9 above)
Comment (Alexey-Ayupov) [reply images/macos/software-report/SoftwareReport.Generator.ps1:261]: Removed

File: images/macos/software-report/SoftwareReport.Generator.ps1 (hunk 8 above)
Comment (Alexey-Ayupov) [reply images/macos/software-report/SoftwareReport.Generator.ps1:43]: No, with -or we will get true on one of the OSes, we need to exclude both. I will rewrite condition in the next manner for better understandability ((-not $os.IsVentura) -and (-not $os.IsVenturaArm64))

File: images/macos/provision/core/xcode.ps1 (hunk 15 above)
Comment (Alexey-Ayupov) [reply images/macos/provision/core/xcode.ps1:21]: done

File: images/macos/provision/core/rubygem.sh (hunk 14 above)
Comment (Alexey-Ayupov) [reply images/macos/provision/core/rubygem.sh:25]: changed to 5.0.0

File: images/macos/provision/core/ruby.sh (hunk 12 above)
Comment (Alexey-Ayupov) [reply images/macos/provision/core/ruby.sh:5]: done

File: images/macos/provision/core/dotnet.sh (hunk 11 above)
Comment (Alexey-Ayupov) [reply images/macos/provision/core/dotnet.sh:51]: done

**Pull request merged**